version.txt

```diff
@@ -1 +0,0 @@
-0.1.6
```

Comment:
Should we remain compatible with the previous version? @ur4t

Reply:
Personally I believe it is no longer useful:

- Inside our project, all usage are replaced.
- In Python packages version info is generated by `setuptools-scm` and stored in `flashinfer._version`.
- For C++ users, usually the whole repo is cloned and they have much more freedom to check versions.

Reply:
Where is the information for the base version obtained from?

Reply:
We plan to support PyPI uploads next week, following PEP 440. The wheel should be uploaded using specifier tags compatible with PEP 440. Wheels that don't need to be uploaded to PyPI can be handled more flexibly.

Reply:
`twine check dist/*whl`

Reply:
> Where is the information for the base version obtained from?

If the repo is on a tag such as `0.1.7`, the wheels will be `flashinfer-0.1.7-cp38-abi3-linux_x86_64.whl`(AOT mode) or `flashinfer-0.1.7-py3-none-any.whl`(JIT mode).

It the repo is not on a tag but a commit after 0.1.7, a local version `+gXXXXXXX` is attached, such as  `flashinfer-0.1.7+gXXXXXXX-cp38-abi3-linux_x86_64.whl`.

It the repo is not clean, `.dYYYYMMDD` is attached to `+gXXXXXXX`.

Additional info like CUDA version and Torch version is collected inside `setup.py` when the wheel is built in AOT mode and attached like `0.1.7(+gXXXXXXX(.dYYYYMMDD)).cu124torch2.5`, but I am afraid it is not proper because `0.1.7.cu124torch2.5` is obviously wrong.

Reply:
`0.1.6+commit_hash.cu124torch2.5` is ok
`0.1.6+commit_hash+cu124torch2.5` is wrong
you can check it with `twine check`

Reply:
> `0.1.6+commit_hash.cu124torch2.5` is ok `0.1.6+commit_hash+cu124torch2.5` is wrong you can check it with `twine check`

Yes. `+` as indicator of local version should only appear once. `0.1.7.cu124torch2.5` is wrong because only `0.1.7.postX` or `0.1.7.devX` is allowed.

AFAIK, Pypi does not allow to upload wheels with local version. So which version of Torch (and underlying CUDA) should we use is also a problem.

Reply:
> AFAIK, Pypi does not allow to upload wheels with local version. So which version of Torch (and underlying CUDA) should we use is also a problem.

I know this. ref https://github.com/flashinfer-ai/flashinfer/pull/666#discussion_r1886921889

Reply:
I have to admit that `setuptools-scm` is good but not suitable for Github actions hierarchy:

- `setuptools-scm` requires more git metadata but `actions/checkout` perfers shallow clone.
- `setuptools-scm` is not that straightforward as `version.txt`.